sdk-react/src/job.ts

```diff
@@ -239,91 +212,25 @@ export const createJob = async <T extends UseAnimaParams = UseAnimaParams>(
-
-    // TODO: For some reason, we receive errors even after the `done` event is triggered.
-    es.addEventListener("error", async (error: ErrorEvent | MessageEvent) => {
-      // Check if the fetch is ok.
-      // If it is't, then the request failed before creating the job and we should terminate the request.
-      const response = await lastFetchResponse;
-      if (!response.ok) {
-        let errorMessage = "";
-        try {
-          errorMessage = await response.text();
-          const errorPayloadJson = JSON.parse(errorMessage);
-          if (errorPayloadJson?.payload?.message) {
-            errorMessage = errorPayloadJson.payload.message;
-          }
-        } catch {}
-
-        reject(
-          new CodegenError({
-            name: "Request failed",
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            reason: errorMessage as any,
-            status: response.status,
-          })
-        );
-        es.close();
-
-        return;
-      }
-
+    es.addEventListener("error", (event) => {
       console.warn(
-        `Experienced error during code generation (attempt ${errorCount + 1} / ${MAX_RETRIES_AFTER_ERROR})`,
-        error
+        `Stream error (attempt ${errorCount + 1} / ${MAX_RETRIES_AFTER_ERROR})`,
+        event
       );
 
-      let errorPayload: StreamMessageByType<"error"> | undefined;
-
-      // If true, then there's no point into keeping the event handler alive for retries.
-      let isUnrecoverableError = false;
-
-      if (error instanceof MessageEvent) {
-        try {
-          errorPayload = JSON.parse(error.data);
-        } catch {}
-      }
-
-      if (
-        errorPayload?.payload?.name === "Task Crashed" ||
-        errorPayload?.payload?.name === "TimeoutError" ||
-        errorPayload?.payload?.name === "Error" ||
-        errorPayload?.payload?.name === "Unknown error"
-      ) {
-        isUnrecoverableError = true;
-      }
-
-      const codegenError = new CodegenError({
-        name: errorPayload?.payload.name ?? "Unknown error",
-        reason: errorPayload?.payload.message ?? "Unknown",
-        status: errorPayload?.payload.status,
-        detail: errorPayload?.payload.detail,
-      });
-
       errorCount++;
+      if (errorCount <= MAX_RETRIES_AFTER_ERROR) return;
 
-      // It could happen that transient network errors cause the "error" event to be triggered temporarily.
-      // In these cases, we ignore the error and retry the request, unless it's an unrecoverable error.
-      let shouldTerminateRequest = false;
-      if (errorCount > MAX_RETRIES_AFTER_ERROR) {
-        console.error("Experienced too many errors, terminating request");
-        shouldTerminateRequest = true;
-      } else if (isUnrecoverableError) {
-        console.error("Experienced unrecoverable error, terminating request");
-        shouldTerminateRequest = true;
-      }
+      const error = new CodegenError({
+        name: "Stream error",
```

Comment:
Just a note to mention these breaking changes in the release notes (renaming the `name` error from `Unknown error` with `Stream error`, which is cleaner).